druid/src/core.rs

```diff
@@ -73,7 +73,7 @@ pub(crate) struct BaseState {
     /// The insets applied to the layout rect to generate the paint rect.
     /// In general, these will be zero; the exception is for things like
     /// drop shadows or overflowing text.
-    paint_insets: Insets,
+    pub(crate) paint_insets: Insets,
```

Comment:
I think it might make sense to have the 'last mouse position' also be in base state? I could imagine it being useful for lots of things, like calculating deltas when you see `MouseMoved`.

Reply:
Having it in base state wouldn't help with this particular PR, because the base state value is going to be stale in `layout`. This could be observed in the timer example but it's not limited to that.

If we end up wanting the last position in base state, then this `layout` pass would just be another update source for the base state's cached value.

Given this, I think it's best to go with [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) here. When we eventually need it in base state, then that's the moment to add it.